### PR TITLE
Add pod annotations support

### PR DESCRIPTION
**Pull request by eyal-moscovici**

### CODE REVIEW

File: git-proxy/values.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Declare variables to be passed into your templates.
 
 replicaCount: 1
-storageService: # s3/gcs/blob/objectstorage
+storageService: "" # s3/gcs/blob/objectstorage
```
Comment (moranCohen26) [inline git-proxy/values.yaml:6]: revert

File: flux2/values.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 git:
   url: http://git-proxy.default.svc.cluster.local
   branch: master
-  path: #./replaceme/
+  path: "" #./replaceme/
```
Comment (moranCohen26) [inline flux2/values.yaml:4]: revert

File: flux2/templates/kustomize-controller.yaml [hunk 3]
```diff
@@ -19,6 +19,9 @@ spec:
       annotations:
         prometheus.io/port: "8080"
         prometheus.io/scrape: "true"
+        {{- if .Values.podAnnotations }}
```
Comment (moranCohen26) [inline flux2/templates/kustomize-controller.yaml:22]: same here, you might have a problem if .Values.podAnnotations will include the prometheus annotations, maybe consider to move the prometheus to the code (getFluxValues) and do the override merge there?

File: flux2/templates/helm-controller.yaml [hunk 4]
```diff
@@ -13,6 +13,9 @@ spec:
   template:
     metadata:
       annotations:
+        {{- if .Values.podAnnotations }}
```
Comment (moranCohen26) [inline flux2/templates/helm-controller.yaml:16]: same here, you might have a problem if .Values.podAnnotations will include the prometheus annotations, maybe consider to move the prometheus to the code (getFluxValues) and do the override merge there?

**Review by moranCohen26 — APPROVED** [review]

File: git-proxy/values.yaml (hunk 1 above)
Comment (eyal-moscovici) [reply git-proxy/values.yaml:6]: This is needed for the template to pass validation:
[ERROR] templates/: template: git-proxy/templates/secret.yaml:18:7: executing "git-proxy/templates/secret.yaml" at <eq .Values.storageService "objectstorage">: error calling eq: incompatible types for comparison

File: flux2/values.yaml (hunk 2 above)
Comment (eyal-moscovici) [reply flux2/values.yaml:4]: done

**Pull request merged**